### PR TITLE
etcd v3.2+ compatibility changes

### PR DESCRIPTION
**Pull request by blakebarnett**
- v3.2+ requires all member URIs are IP addresses

I'm itching to refactor this, but it works and scratched the itch temporarily.

### CODE REVIEW

**Comment by unguiculus** [comment]
Please fix conflicts.

File: incubator/etcd/templates/etcd-statefulset.yaml [hunk 1]
```diff
@@ -120,16 +122,16 @@ spec:
...
-                ETCDCTL_ENDPOINT=$(eps) etcdctl member update ${member_id} http://${HOSTNAME}.${SET_NAME}:2380
-                exec etcd --name ${HOSTNAME} \
-                    --listen-peer-urls http://${HOSTNAME}.${SET_NAME}:2380 \
-                    --listen-client-urls http://${HOSTNAME}.${SET_NAME}:2379,http://127.0.0.1:2379 \
-                    --advertise-client-urls http://${HOSTNAME}.${SET_NAME}:2379 \
+                ETCDCTL_ENDPOINT=$(eps) etcdctl member update ${member_id} "http://$MY_POD_IP:2380"
+                exec etcd --name "$MY_POD_IP" \
+                    --listen-peer-urls "http://$MY_POD_IP:2380" \
+                    --listen-client-urls "http://$MY_POD_IP:2379,http://127.0.0.1:2379" \
+                    --advertise-client-urls "http://$MY_POD_IP:2379" \
```
Comment (xiang90) [inline incubator/etcd/templates/etcd-statefulset.yaml:129]: you can advertise hostname/dns. only listen addr is restricted to IP.

**Comment by blakebarnett** [comment]
I've since dropped this in favor of `etcd-operator`, I couldn't make a cluster deployed by this chart survive a partition reliably.

**Pull request closed without merging**